lisp/ess-mode.el

```diff
@@ -317,9 +315,16 @@ indentation style. See `ess-style-alist' for predefined styles."
   (delq t comint-dynamic-complete-functions)
   (set (make-local-variable 'comint-completion-addsuffix)
        (cons "/" ""))
-
+  (add-hook 'hack-local-variables-hook #'ess--set-style-in-buffer nil t)
   (add-hook 'ess-idle-timer-functions 'ess-synchronize-dirs nil 'local))
 
+(defun ess--set-style-in-buffer ()
+  "Set `ess-style' taking into account file and directory local variables.
+This is added to `hack-local-variables-hook'."
```

Comment:
Cannot this one be done in ess-mode directly? Why to bother with `hack-local-variables-hook`?

Reply:
Local variables are set after the major modes, so we'll miss them if it's inside `ess-mode`.